python/modules/IcePy/BatchRequestInterceptor.cpp

```diff
@@ -133,55 +133,22 @@ static PyMethodDef BatchRequestMethods[] = {
-        sizeof(BatchRequestObject),                       /* tp_basicsize */
-        0,                                                /* tp_itemsize */
-        /* methods */
-        reinterpret_cast<destructor>(batchRequestDealloc), /* tp_dealloc */
-        0,                                                 /* tp_print */
-        0,                                                 /* tp_getattr */
-        0,                                                 /* tp_setattr */
-        0,                                                 /* tp_reserved */
-        0,                                                 /* tp_repr */
-        0,                                                 /* tp_as_number */
-        0,                                                 /* tp_as_sequence */
-        0,                                                 /* tp_as_mapping */
-        0,                                                 /* tp_hash */
-        0,                                                 /* tp_call */
-        0,                                                 /* tp_str */
-        0,                                                 /* tp_getattro */
-        0,                                                 /* tp_setattro */
-        0,                                                 /* tp_as_buffer */
-        Py_TPFLAGS_DEFAULT,                                /* tp_flags */
-        0,                                                 /* tp_doc */
-        0,                                                 /* tp_traverse */
-        0,                                                 /* tp_clear */
-        0,                                                 /* tp_richcompare */
-        0,                                                 /* tp_weaklistoffset */
-        0,                                                 /* tp_iter */
-        0,                                                 /* tp_iternext */
-        BatchRequestMethods,                               /* tp_methods */
-        0,                                                 /* tp_members */
-        0,                                                 /* tp_getset */
-        0,                                                 /* tp_base */
-        0,                                                 /* tp_dict */
-        0,                                                 /* tp_descr_get */
-        0,                                                 /* tp_descr_set */
-        0,                                                 /* tp_dictoffset */
-        0,                                                 /* tp_init */
-        0,                                                 /* tp_alloc */
-        reinterpret_cast<newfunc>(batchRequestNew),        /* tp_new */
-        0,                                                 /* tp_free */
-        0,                                                 /* tp_is_gc */
+        PyVarObject_HEAD_INIT(nullptr, 0)
```

Comment:
Using designated initializers make initialization clear, and safer. The comments in the previous version refer to members that no longer exits like `tp_print` and the `0` assignment was really to a different member.